### PR TITLE
[RA1 Ch02] APIs: requirement added per service (#1621)

### LINKED ISSUE #1621: [RA1- Ch02] APIs: requirements for all OpenStack Mandatory Services APIs
In section 2.3.4 "Interfaces & APIs requirements", we should add a dedicated requirement for each OpenStack mandatory service. Currently, only "Horizon" and "Placement" have a dedicated requirement and req.int.api.01 is too generic to cover all the other services

### PR DESCRIPTION
**Pull request by collivier**
* [RA1 Ch02] APIs: requirement added per service

fixes #1621
Section 2.3.4 APIs
Requirements added for each OpenStack Mandatory Service

* Update doc/ref_arch/openstack/chapters/chapter02.md

Co-authored-by: Pankaj Goyal <52107136+pgoyal01@users.noreply.github.com>

* 2.3.4 Update per Pankaj's comment

"CNTT conformance" removed

* Typo

Co-authored-by: Pankaj Goyal <52107136+pgoyal01@users.noreply.github.com>
(cherry picked from commit 3c632cd8ffce7120ca624dbe26fcc30e7cf35f09)

### CODE REVIEW

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 1]
```diff
@@ -105,11 +105,16 @@ Traceability to Reference Model.
...
-| `req.int.api.02` | API | The Architecture **must** provide GUI access to tenant facing cloud platform core services. | [RA-1 4.3.1.9 "Horizon"](./chapter04.md#4319-horizon) |
-| `req.int.api.03` | API | The Architecture **must** provide APIs needed to discover and manage Cloud Infrastructure resources. | [RA-1 5.2.7. "Placement"](./chapter05.md#527-placement) |
-| `req.int.api.04` | API | The Architecture must expose the latest version and microversion of the APIs for the given CNTT OpenStack release for each of the OpenStack core services | [RA-1 5.2 Core OpenStack Services APIs](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter05.md#52-core-openstack-services-apis) |
-
+| `req.int.api.01` | API | The Architecture **must** provide APIs to access to the authentication service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.1 "Keystone"](./chapter05.md#521-keystone) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:108]: Typo
```suggestion
| `req.int.api.01` | API | The Architecture **must** provide APIs to access the authentication service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.1 "Keystone"](./chapter05.md#521-keystone) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 2]
```diff
@@ -105,11 +105,16 @@ Traceability to Reference Model.
...
+| `req.int.api.02` | API | The Architecture **must** provide APIs to access to the image management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.2 "Glance"](./chapter05.md#522-glance) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:109]: Typo
```suggestion
| `req.int.api.02` | API | The Architecture **must** provide APIs to access the image management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.2 "Glance"](./chapter05.md#522-glance) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 3]
```diff
@@ -105,11 +105,16 @@ Traceability to Reference Model.
...
+| `req.int.api.03` | API | The Architecture **must** provide APIs to access to the block storage management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.3 "Cinder"](./chapter05.md#523-cinder) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:110]: Typo
```suggestion
| `req.int.api.03` | API | The Architecture **must** provide APIs to access the block storage management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.3 "Cinder"](./chapter05.md#523-cinder) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 4]
```diff
@@ -105,11 +105,16 @@ Traceability to Reference Model.
...
+| `req.int.api.04` | API | The Architecture **must** provide APIs to access to the object storage management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.4 "Swift"](./chapter05.md#524-swift) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:111]: Typo
```suggestion
| `req.int.api.04` | API | The Architecture **must** provide APIs to access the object storage management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.4 "Swift"](./chapter05.md#524-swift) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 5]
```diff
@@ -105,11 +105,16 @@ Traceability to Reference Model.
...
+| `req.int.api.05` | API | The Architecture **must** provide APIs to access to the network management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.5 "Neutron"](./chapter05.md#525-neutron) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:112]: Typo
```suggestion
| `req.int.api.05` | API | The Architecture **must** provide APIs to access the network management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.5 "Neutron"](./chapter05.md#525-neutron) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 6]
```diff
@@ -105,11 +105,16 @@ Traceability to Reference Model.
...
+| `req.int.api.06` | API | The Architecture **must** provide APIs to access to the compute resources management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.6 "Nova"](./chapter05.md#526-nova) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:113]: Typo
```suggestion
| `req.int.api.06` | API | The Architecture **must** provide APIs to access the compute resources management service and the associated mandatory features detailed in chapter 5. | [RA-1 5.2.6 "Nova"](./chapter05.md#526-nova) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 7]
```diff
@@ -105,11 +105,16 @@ Traceability to Reference Model.
...
+| `req.int.api.07` | API | The Architecture **must** provide GUI access to tenant facing cloud platform core services. | [RA-1 4.3.1.9 "Horizon"](./chapter04.md#4319-horizon) |
+| `req.int.api.08` | API | The Architecture **must** provide APIs needed to discover and manage Cloud Infrastructure resources. | [RA-1 5.2.7. "Placement"](./chapter05.md#527-placement) |
+| `req.int.api.09` | API | The Architecture **must** provide APIs to access to the orchestration service. | [RA-1 5.2.8 "Heat"](./chapter05.md#528-heat) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:116]: Typo
```suggestion
| `req.int.api.09` | API | The Architecture **must** provide APIs to access the orchestration service. | [RA-1 5.2.8 "Heat"](./chapter05.md#528-heat) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md (hunk 2 above)
Comment (collivier) [reply doc/ref_arch/openstack/chapters/chapter02.md:109]: Well catch ! see https://github.com/cntt-n/CNTT/pull/1945
The second change would be cherry-picked to complete this one (else master would diverge from baldy)

File: doc/ref_arch/openstack/chapters/chapter02.md (hunk 4 above)
Comment (collivier) [reply doc/ref_arch/openstack/chapters/chapter02.md:111]: Well catch ! see #1945
The second change would be cherry-picked to complete this one (else master would diverge from baldy)

File: doc/ref_arch/openstack/chapters/chapter02.md (hunk 5 above)
Comment (collivier) [reply doc/ref_arch/openstack/chapters/chapter02.md:112]: Well catch ! see #1945
The second change would be cherry-picked to complete this one (else master would diverge from baldy)

File: doc/ref_arch/openstack/chapters/chapter02.md (hunk 7 above)
Comment (collivier) [reply doc/ref_arch/openstack/chapters/chapter02.md:116]: Well catch ! see #1945
The second change would be cherry-picked to complete this one (else master would diverge from baldy)

File: doc/ref_arch/openstack/chapters/chapter02.md (hunk 6 above)
Comment (collivier) [reply doc/ref_arch/openstack/chapters/chapter02.md:113]: Well catch ! see #1945
The second change would be cherry-picked to complete this one (else master would diverge from baldy)

File: doc/ref_arch/openstack/chapters/chapter02.md (hunk 1 above)
Comment (collivier) [reply doc/ref_arch/openstack/chapters/chapter02.md:108]: Well catch ! see #1945
The second change would be cherry-picked to complete this one (else master would diverge from baldy)

**Comment by collivier** [comment]
Will be completed by #1945

**Review by karinesevilla — APPROVED** [review]

**Pull request merged**